src/robot/reporting/jsbuildingcontext.py

```diff
@@ -47,6 +48,7 @@ def string(self, string, escape=True, attr=False):
             string = (html_escape if not attr else attribute_escape)(string)
         return self._strings.add(string)
 
+    @functools.lru_cache
```

Comment:
The things to keep in mind here:
- Using these automatic cachings means that the function should be side-effect free or things will go awry.
- The cache size can be tuned. My test data being just a big dumb with very few separate keywords called makes it very easy to cache.
  - Should investigate memory impact.
  - Investigate performance impact of smaller / lower cache value (the cache is dict, but even being logarithmic if it grows large it can have negative performance impact.)

Reply:
But on a general level just this code change made `rebot` runs on my machine 10-15% faster (from 18 seconds to 16 without flattenkeywords and from 9 to 7-8 with flattenkeywords).

Reply:
Plr

Reply:
```
--- 5.0b2.dev1 master ---

(.venv)  ~/C/t/rf-5.0-iteration-longrun (master)> python benchmark.py
Time for simple-million-while-output.xml
146.158634542
Time for ./rfbrowser-output.xml
0.18507958300000382
Time for ./rf-full-suite-output.xml
44.82903858399999


--- Switch branch to faster-jsbuildingcontext ---

Time for simple-million-while-output.xml
134.029772417
Time for ./rfbrowser-output.xml
0.164100042
Time for ./rf-full-suite-output.xml
38.828153250000014
```

These are summed (maybe averages with listed deviation would be better) times of 10 runs of each dataset with `timeit`.

Increase seems to be around 10% both with small and larger output.xml:s. I did not do any memory profiling yet.